src/common/hobject.cc

```diff
@@ -288,11 +301,15 @@ void ghobject_t::generate_test_instances(list<ghobject_t*>& o)
 
 ostream& operator<<(ostream& out, const ghobject_t& o)
 {
+  if (o == ghobject_t())
+    return out << "GHMIN";
+  if (o.is_max())
+    return out << "GHMAX";
+  if (o.shard_id != shard_id_t::NO_SHARD)
+    out << o.shard_id << ":";
   out << o.hobj;
-  if (o.generation != ghobject_t::NO_GEN ||
-      o.shard_id != shard_id_t::NO_SHARD) {
-    assert(o.shard_id != shard_id_t::NO_SHARD);
-    out << "/" << o.generation << "/" << (unsigned)(o.shard_id);
+  if (o.generation != ghobject_t::NO_GEN) {
+    out << "/" << (unsigned)(o.shard_id);
```

Comment:
On Thu, 8 Jan 2015, David Zafman wrote:
> 
> In src/common/hobject.cc:
> 
> >    out << o.hobj;
> > -  if (o.generation != ghobject_t::NO_GEN ||
> > -      o.shard_id != shard_id_t::NO_SHARD) {
> > -    assert(o.shard_id != shard_id_t::NO_SHARD);
> > -    out << "/" << o.generation << "/" << (unsigned)(o.shard_id);
> > +  if (o.generation != ghobject_t::NO_GEN) {
> > +    out << "/" << (unsigned)(o.shard_id);
> 
> I think you meant o.generation here.

omg that would have been a nightmare to debug later